includes/Services/FlowService.php

```diff
@@ -0,0 +1,203 @@
+		$flow_data_fixes = Flows::get_fixes();
+		$exception_list = Flows::get_exception_list();
+		$updated_flow_data = array();
+		foreach ($default_flow_data as $key => $value)
+		{ 
+			// To align the datatype of the respective values with the one set in the blueprint
+			if(strcmp(gettype($value), gettype($flow_data[$key])) != 0) {
+				$updated_flow_data[$key] = $value;
+				continue;
+			}
+
+			// Any Key renamed is updated in the database with NewKey and the value from the OldKey is retained or not based on retain_existing_value
+			if(sizeof($flow_data_fixes) > 0) {
+				foreach ($flow_data_fixes as $index=>$fix) {
+					if (array_key_exists($fix['old_key'], $flow_data) && array_key_exists($fix['new_key'], $default_flow_data)) {
+						($fix['retain_existing_value'])?
+							$flow_data[$fix['new_key']] = $flow_data[$fix['old_key']]
+							: $flow_data[$fix['new_key']] = $default_flow_data[$fix['new_key']];
+						unset($flow_data[$fix['old_key']]);
+					}	
+				}
+			}
+
+			if( ! array_key_exists($key, $flow_data)) {
+				$updated_flow_data[$key] = $value;
+				continue;
+			}
+
+			// Accepts the value of Exception List keys as is from the database OR non array OR empty array values
+			if(isset($exception_list[$key]) || ! is_array($value) || (is_array($value) && sizeof($value) === 0)) {
+				$updated_flow_data[$key] = $flow_data[$key];
+				continue;
+			}
+
+			// To handle Indexed Arrays gracefully
+			if (count(array_filter(array_keys($value), 'is_string')) === 0) {
+				// To check if an Indexed Array is further Nested or Not
+				foreach($value as $index_key => $index_value) {
+					// For Indexed Arrays having Non Associative Array Values
+					(!is_array($index_value) && isset($flow_data[$key]))? 
+						$updated_flow_data[$key] = $flow_data[$key] 
+						: $updated_flow_data[$key] = $value;
+				}
+				continue;
+			} 
+
+			// To handle Associative Arrays gracefully
+			$updated_flow_data[$key] = self::update_flow_data_recursive($value, $flow_data[$key]);
+		}
+		return $updated_flow_data;
+	}
+
+	/*
+	 * function to update the Database recursively based on Values opted or entered by the User
+	 */	
+	private static function update_post_call_data_recursive(&$flow_data, $params) {
+		$exception_list = Flows::get_exception_list();
+
+		foreach ($flow_data as $key => $value)
+		{ 		
+			if (! array_key_exists($key, $params)) {
+				$flow_data[$key] = $value;	
+				continue;
+			}
+
+			// Accepts the value of Exception List keys as is from the database
+			if($exception_list && isset($exception_list[$key])) {
+				$flow_data[$key] = $params[$key];
+				continue;
+			}
+
+			// Error thrown if the datatype of the parameter does not match
+			if(strcmp(gettype($value), gettype($params[$key])) != 0) {
+				return new \WP_Error(
+					'wrong_param_type_provided',
+					"Wrong Parameter Type Provided : ". $key. ' => '. gettype($params[$key]) . '. Expected: ' . gettype($value),
+					array( 'status' => 404 )
```

Comment:
```suggestion
					array( 'status' => 400 )
```
404 is used if something is not found https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses